src/Microsoft.DotNet.Interactive.Jupyter/ConnectJupyterKernelCommand.cs

```diff
@@ -69,7 +69,10 @@ public ConnectJupyterKernelCommand AddConnectionOptions(IJupyterKernelConnection
         var localName = commandLineContext.ParseResult.GetValueForOption(KernelNameOption);
 
         var kernel = await connector?.CreateKernelAsync(localName);
-        kernel?.RegisterForDisposal(connection);
+        if (connection is IDisposable disposableConnection)
+        {
+            kernel?.RegisterForDisposal(disposableConnection);
```

Comment:
What can lead to kernel being null?

Reply:
There shouldn't be a case of kernel being null. I think this was just guarding but I think probably we would want to throw in that case. Will fix separately.

Reply:
fixed in follow up PR: https://github.com/dotnet/interactive/pull/2892